File: frontend/src/lib/types/job.ts
import { PageRequest } from "./pagination";

export interface Job {
  id: string;
  from_site: string;
  job_url: string;
  logo_url: string;
  job_name: string;
  job_level?: string;
  company_name: string;
  company_type?: string;
  company_address?: string;
  company_description?: string;
  job_type?: string;
  skills: string;
  location?: string;
  date_posted?: string;
  job_description: string;
  job_requirement: string;
  benefit?: string;
  is_analyze: boolean;
  is_favorite: boolean;
}

export interface FavoriteJobRequest {
  job_id: string;
  user_id: string;
  is_analyze?: boolean;
  is_generated_resume?: boolean;
  is_favorite: boolean;
}

export interface JobAnalytic extends Job {
  id: string;
  job_id: string;
  match_overall: number;
  match_experience: number;
  match_skills: number;
  weaknesses: string;
  strengths: string;
  overall_assessment: string;
  strength_details: string;
  weakness_concerns: string;
  recommendations: string;
  questions: string;
  roadmap: string;
  conclusion: string;
}

export interface JobFavoriteResponse extends Job {
  job_analytics?: JobAnalytic;
}

export interface JobSearchRequest extends PageRequest {
  query: string;
  roles?: string[];
  levels?: string[];
}

export interface JobAnalyticSearchRequest extends PageRequest {
  search?: string;
}

File: backend/app/convert/job.py
from typing import Optional

from app.model.favorite_job import FavoriteJob
from app.model.job import Job
from app.model.job_analytic import JobAnalytic
from app.schema.job_analytic_schema import JobAnalyticResponse
from app.schema.job_schema import JobFavoriteResponse

def to_favorite_job_response(
    job: Job,
    fav: FavoriteJob,
    analytic: Optional[JobAnalytic]
) -> JobFavoriteResponse:
    analytic_resp = (
        JobAnalyticResponse.model_validate(analytic)
        if analytic
        else None
    )
    
    return JobFavoriteResponse(
        id=job.id,
        job_url=job.job_url or "",
        logo_url=job.logo_url or "",
        job_name=job.job_name or "",
        from_site=job.from_site or "",
        company_name=job.company_name or "",
        company_type=job.company_type,
        company_address=job.company_address,
        company_description=job.company_description,
        job_type=job.job_type,
        skills=job.skills or "",
        location=job.location,
        date_posted=job.date_posted,
        salary=job.salary,
        job_description=job.job_description or "",
        is_analyze=fav.is_analyze if fav else False,
        is_favorite=fav.is_favorite if fav else False,
        job_analytics=analytic_resp,
    )
